Kernel/KBOComparator.cpp

```diff
@@ -92,10 +92,10 @@ void KBOComparator::makeReady()
 
     // we only care about the non-zero weights and counts
     bool varInbalance = false;
-    auto state = kbo._state;
+    auto state = &*kbo._state;
 #if VDEBUG
     // we make sure kbo._state is not used while we're using it
-    kbo._state = nullptr;
+    auto __state = std::move(kbo._state);
```

Comment:
I don't think this preserves the old behaviour - agree with the move, but shoudn't `kbo._state` still be assigned `nullptr`?

Reply:
Isn't moving out of a `unique_ptr` the same thing as nulling it? It's then a pointer that cannot be accessed, right?

Reply:
Yes, you're right - apparently the standard specifies the moved-from state of `unique_ptr`.